DataFormats/ParticleFlowReco/src/PFRecHit.cc

```diff
@@ -2,8 +2,50 @@
 using namespace reco;
 using namespace std;
 
+#include "vdt/vdtMath.h"
+#include "Math/GenVector/etaMax.h"
+
 const unsigned    PFRecHit::nCorners_ = 4;
 
+namespace {
+  
+  // an implementation of Eta_FromRhoZ of root libraries using vdt
+  template<typename Scalar>
+  inline Scalar Eta_FromRhoZ_fast(Scalar rho, Scalar z) {    
+    using namespace ROOT::Math;
+    // value to control Taylor expansion of sqrt
+    constexpr Scalar big_z_scaled =
+      std::pow(std::numeric_limits<Scalar>::epsilon(),static_cast<Scalar>(-.25));
```

Comment:
bah
>>> pow(1.e-7,-0.25)
56.23413251903491
>>> 1./sqrt(sqrt(1.e-7))
56.2341325190349
most probably 50 is good enough
in any case I do not see how a PFRecHit can stay at eta>50 ....